README.md

```diff
@@ -1,2 +1,221 @@
+
+//take 5 values and print response
+stream.take(5).collect { payload: Payload ->
+    println(payload.data.readText())
+}
+```
+
+### Server WebSocket with CIO ktor engine
+
+```kotlin
+//create ktor server
+embeddedServer(CIO) {
+    install(RSocketServerSupport) {
+        //configure rSocket server
+    }   
+    //configure routing
+    routing {
+        //configure route `url:port/rsocket`
+        rSocket("rsocket") {
+            RSocketRequestHandler {
+                //handler for request/response
+                requestResponse = { request: Payload ->
+                    //... some work here
+                    delay(500) // work emulation
+                    Payload("data", "metadata")
+                }         
+                //handler for request/stream      
+                requestStream = { request: Payload ->
+                    flow {
+                        repeat(1000) { i -> 
+                            emit(Payload("data: $i"))
+                        }               
+                    }        
+                }           
+            }       
+        }       
+    }
+}.start(true)
+```
+
+### More examples:
+
+- [interactions](examples/interactions) - contains usages of some supported functions
+- [multiplatform-chat](examples/multiplatform-chat) - chat implementation with JVM server and JS/JVM client with shared classes
+and serializing data using [kotlinx.serialization](https://github.com/Kotlin/kotlinx.serialization)  
+- [nodejs-tcp-transport](examples/nodejs-tcp-transport) - implementation of TCP transport for nodejs 
+
+## Reactive Streams Semantics
+
+From [RSocket protocol](https://github.com/rsocket/rsocket/blob/master/Protocol.md#reactive-streams-semantics):
+
+    Reactive Streams semantics are used for flow control of Streams, Subscriptions, and Channels. 
+    This is a credit-based model where the Requester grants the Responder credit for the number of PAYLOADs it can send. 
+    It is sometimes referred to as "request-n" or "request(n)".
+
+`kotlinx.coroutines` doesn't truly support `request(n)` semantic, but it has `Flow.buffer(n)` operator
+which can be used to archive similar possibilities:
+
+Example:
+
+```kotlin
+//assume we have client
+val client: RSocket = TODO()
+
+//and stream
+val stream: Flow<Payload> = client.requestStream(Payload("data"))
+
+//now we can use buffer to tell underlying transport to request values in chunks
+val bufferedStream: Flow<Payload> = stream.buffer(10) //here buffer is 10, if `buffer` operator is not used buffer is by default 64
+
+//now you can collect as any other `Flow`
+//just after collection first request for 10 elements will be sent
+//after 10 elements collected, 10 more elements will be requested, and so on
+bufferedStream.collect { payload: Payload ->
+    println(payload.data.readText())
+}
+``` 
+
+## Bugs and Feedback
+
+For bugs, questions and discussions please use the [Github Issues](https://github.com/rsocket/rsocket-kotlin/issues).
```

Comment:
Slack?

Reply:
I thought that rsocket slack isn't public, if so, does it make sense?